test/entt/entity/polymorphic.cpp

```diff
@@ -0,0 +1,433 @@
+
+    ASSERT_EQ(entt::algorithm::poly_count<cube>(reg), 1);
+    ASSERT_EQ(entt::algorithm::poly_count<cat>(reg), 2);
+    ASSERT_EQ(entt::algorithm::poly_count<dog>(reg), 2);
+    ASSERT_EQ(entt::algorithm::poly_count<sphere>(reg), 2);
+    ASSERT_EQ(entt::algorithm::poly_count<animal>(reg), 4);
+    ASSERT_EQ(entt::algorithm::poly_count<shape>(reg), 3);
+    ASSERT_EQ(entt::algorithm::poly_count<fat_cat>(reg), 1);
+}
+
+
+TEST(PolyEach, Functionalities) {
+    entt::registry reg;
+
+    entt::entity entity1 = reg.create();  // components: cat(payload=1), dog(payload=2)
+    entt::entity entity2 = reg.create();  // components: dog(payload=2), cube(payload=3), sphere(payload=4)
+    entt::entity entity3 = reg.create();  // components: fat_cat(both payloads = 5)
+    reg.emplace<cat>(entity1).animal_payload = 1;
+    reg.emplace<dog>(entity1).animal_payload = 2;
+    reg.emplace<dog>(entity2).animal_payload = 2;
+    reg.emplace<cube>(entity2).shape_payload = 3;
+    reg.emplace<sphere>(entity2).shape_payload = 4;
+    auto& c = reg.emplace<fat_cat>(entity3);
+    c.animal_payload = c.shape_payload = 5;
+
+    {
+        int count = 0;
+        entt::algorithm::poly_each<cube>(reg, [&] (entt::entity e, cube& s) -> void {
+            ASSERT_TRUE(e == entity2);
+            ASSERT_TRUE(s.shape_payload == 3);
+            count++;
+        });
+        ASSERT_EQ(count, 1);
+    }
+
+    {
+        int count = 0;
+        entt::algorithm::poly_each<animal>(reg, [&] (entt::entity e, animal& a) -> void {
+            ASSERT_TRUE(e == entity1 || e == entity2 || e == entity3);
+            if (e == entity1) {
+                ASSERT_TRUE(a.animal_payload == 1 || a.animal_payload == 2);
+                ASSERT_TRUE(a.animal_payload != 1 || a.name() == "cat");
+                ASSERT_TRUE(a.animal_payload != 2 || a.name() == "dog");
+            } else if (e == entity2) {
+                ASSERT_TRUE(a.animal_payload == 2 && a.name() == "dog");
+            } else if (e == entity3) {
+                ASSERT_TRUE(a.animal_payload == 5 && a.name() == "cat");
+            }
+            count++;
+        });
+        ASSERT_EQ(count, 4);
+    }
+
+    {
+        int count = 0;
+        entt::algorithm::poly_each<shape>(reg, [&] (entt::entity e, shape& s) -> void {
+            ASSERT_TRUE(e == entity2 || e == entity3);
+            if (e == entity2) {
+                ASSERT_TRUE(s.shape_payload == 3 || s.shape_payload == 4);
+                ASSERT_TRUE(s.shape_payload != 3 || s.draw() == "cube");
+                ASSERT_TRUE(s.shape_payload != 4 || s.draw() == "sphere");
+            } else if (e == entity3) {
+                ASSERT_TRUE(s.shape_payload == 5 && s.draw() == "sphere");
+            }
+            count++;
+        });
+        ASSERT_EQ(count, 3);
+    }
+
+    {
+        entt::algorithm::poly_each<animal*>(reg, [&] (entt::entity e, [[maybe_unused]] animal* a) -> void {
+            ASSERT_TRUE(false);
```

Comment:
Codecov fails here, is there is a way to check, if the function is never called, without lowering the code coverage?

Reply:
I think putting everything on a single line is enough to trick codecov. Maybe use `FAIL()` too.